operators/multiclusterobservability/controllers/multiclusterobservability/analytics/rightsizing_namespace_configmap.go

```diff
@@ -0,0 +1,194 @@
+	}
+
+	return nil
+}
+
+func GetDefaultRSNamespaceConfig() map[string]string {
+	// get default config data with PrometheusRule config and placement config
+
+	var ruleConfig RSPrometheusRuleConfig
+	ruleConfig.NamespaceFilterCriteria.ExclusionCriteria = []string{"openshift.*"}
+	ruleConfig.RecommendationPercentage = rsDefaultRecommendationPercentage
+
+	placement := &clusterv1beta1.Placement{
+		Spec: clusterv1beta1.PlacementSpec{
+			Predicates: []clusterv1beta1.ClusterPredicate{},
+			Tolerations: []clusterv1beta1.Toleration{
+				{
+					Key:      "cluster.open-cluster-management.io/unreachable",
+					Operator: clusterv1beta1.TolerationOpExists,
+				},
+				{
+					Key:      "cluster.open-cluster-management.io/unavailable",
+					Operator: clusterv1beta1.TolerationOpExists,
+				},
+			},
+		},
+	}
+
+	return map[string]string{
+		"prometheusRuleConfig":   FormatYAML(ruleConfig),
+		"placementConfiguration": FormatYAML(placement),
+	}
+}
+
+// GetRightSizingConfigData extracts and unmarshals the data from the ConfigMap into RightSizingConfigData
+func GetRightSizingConfigData(cm *corev1.ConfigMap) (RSNamespaceConfigMapData, error) {
+	var configData RSNamespaceConfigMapData
+
+	// Unmarshal namespaceFilterCriteria
+	if err := yaml.Unmarshal([]byte(cm.Data["prometheusRuleConfig"]), &configData.PrometheusRuleConfig); err != nil {
+		log.Error(err, "Failed to unmarshal prometheusRuleConfig")
+		return configData, fmt.Errorf("Failed to unmarshal prometheusRuleConfig: %v", err)
+	}
+
+	// Unmarshal placementConfiguration
+	if cm.Data["placementConfiguration"] != "" {
+		if err := yaml.Unmarshal([]byte(cm.Data["placementConfiguration"]), &configData.PlacementConfiguration); err != nil {
+			log.Error(err, "Failed to unmarshal placementConfiguration")
+			return configData, fmt.Errorf("Failed to unmarshal placementConfiguration: %v", err)
+		}
+	}
+
+	// Log or process the `configData` as needed
+	log.Info("RS - ConfigMap Data successfully unmarshalled")
+
+	return configData, nil
+}
+
+func GetNamespaceRSConfigMapPredicateFunc(c client.Client) predicate.Funcs {
+	log.Info("RS - Watch for ConfigMap events set up started")
+
+	processConfigMap := func(cm *corev1.ConfigMap) bool {
+		configData, err := GetRightSizingConfigData(cm)
+		if err != nil {
+			log.Error(err, "Failed to extract RightSizingConfigData")
+			return false
+		}
+
+		// Apply changes based on the config map
+		if err := applyRSNamespaceConfigMapChanges(context.Background(), c, configData); err != nil {
+			log.Error(err, "Failed to apply RS Namespace ConfigMap Changes")
+			return false
+		}
+		return true
+	}
+
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			if e.Object.GetName() == rsConfigMapName && e.Object.GetNamespace() == config.GetDefaultNamespace() {
+				if cm, ok := e.Object.(*corev1.ConfigMap); ok {
+					return processConfigMap(cm)
+				}
+			}
+			return true
```

Comment:
Shouldn't be here `false`?

Reply:
Updated the return as false.

Reply:
As we have two predicate functions, we are taking approach of **Loosen Restrictions – Return true When Event Not Relevant**. 

Also `e2e-kind` is failing with returning `false`, so keeping it with `true`